frontend/src/components/Scratch/RenderTabs.tsx

```diff
@@ -0,0 +1,135 @@
+import { useRef } from "react"
+
+import { Compilation, Scratch } from "../../lib/api"
+import CompilerOpts from "../compiler/CompilerOpts"
+import Diff from "../Diff"
+import Editor from "../Editor"
+import { EditorInstance } from "../Editor/MonacoEditor"
+import ScoreBadge from "../ScoreBadge"
+import { Tab } from "../Tabs"
+
+import AboutScratch from "./AboutScratch"
+import styles from "./Scratch.module.scss"
+
+type ScratchTab = LeftScratchTab | RightScratchTab
+
+function renderTabs(tabs: {[i: number]: JSX.Element}, filter?: Array<ScratchTab>) {
+    if (!filter)
+        return Object.values(tabs)
+    else
+        return Object.keys(tabs).filter(key => filter.includes(+key)).map(key => tabs[key])
+}
+
+export enum LeftScratchTab {
+    SOURCE_CODE,
+    ABOUT,
+    CONTEXT,
+    SETTINGS,
+}
+
+export enum RightScratchTab {
+    DIFF,
+}
+
+/**
+ * Renders the left tabs of the scratch
+ * @param {Array<LeftScratchTab>} [filter=undefined] The tabs you want to render
```

Comment:
Can you explain this doc comment to me?

Reply:
Maybe "render" was the wrong wording for this, but basically "Returns the left tabs of the scratch layout".
The second line also says that filter can be undefined (filter=undefined), i think thats the way you do it in JSDoc.